polaris-react/src/components/IndexTable/IndexTable.module.scss

```diff
@@ -159,6 +159,10 @@ $loading-panel-height: 53px;
     cursor: auto;
   }
 
+  &.TableRow-borderless:not(:first-child) {
```

Comment:
It is likely an issue with consumer implementation and not this PR, however, leaving it here just in case as it's different from current production.

The first table row results in a double border on mobile.

<img width="1911" alt="Screenshot 2024-03-07 at 9 53 48 AM" src="https://github.com/Shopify/polaris/assets/15054990/9d0ba4b9-f20e-4aef-bbc2-ec36c09c618e">

Reply:
Yes, that was an issue on consumer level, fixed in web PR https://github.com/Shopify/web/pull/120568